.travis.yml

```diff
@@ -2,20 +2,39 @@ language: python
 python:
     - "2.7"
     - "3.6"
-    - "nightly"
+
 # Sudo is required so that steup.py can be tested (TODO: investigate using
 # a virtualenv instead?)
 sudo: required
+addons:
+  apt:
+    packages:
+    - realpath # required by script
+    - cabal-install
+    - ghc
 os: linux
 install:
   - pip install pycodestyle setuptools
   - gem install travis --no-rdoc --no-ri
 
+  # https://github.com/koalaman/shellcheck/wiki/TravisCI
+  - curl -L "https://raw.githubusercontent.com/Lin-Buo-Ren/Utilities-for-Travis-CI/master/Build%20and%20Setup%20ShellCheck's%20Latest%20Release.bash" -o setup_shellcheck.bash
+  - chmod +x setup_shellcheck.bash
+  - ./setup_shellcheck.bash --without-root
+
```

Comment:
See above. I would prefer to use [this shellcheck](https://github.com/koalaman/shellcheck) as it appears to have been further developed than the one used here and can be installed with `apt`. Note also that the shell check used here is specific to bash and does NOT enforce POSIX compliance.

Reply:
Looking at the [https://github.com/koalaman/shellcheck](repository), it appears to be a wrapper around the apt package mentioned. I would suggest that if we want this wrapper, we should add it as a separate file to the project instead of a dependency.